gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/CodeGenerator.js

```diff
@@ -0,0 +1,46 @@
+/*!
+Copyright 2014 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion");
+var crypto = require("crypto");
+
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.oauth2.codeGenerator", {
```

Comment:
The gpii.oauth2.codeGenerator component generates OAuth 2 authentication codes and access tokens. The functionality was originally part of the AuthorizationService and I moved the work to this new component when I added unit tests for the AuthorizationService.

The code generation is using the Nodejs crypto module and I wanted to write tests that would run in the browser, so I moved this code out into a separate component.

Does this seem reasonable? Would there be a better solution?

None of the reasoning for this components existence is visible in the code -- would it be worthwhile to document the above in a comment on the gpii.oauth2.codeGenerator component?